node/silkworm/downloader/internals/body_sequence.cpp

```diff
@@ -14,10 +14,347 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+#include <silkworm/chain/difficulty.hpp>
+#include <silkworm/common/log.hpp>
+#include <silkworm/consensus/base/engine.hpp>
+
 #include "body_sequence.hpp"
+#include "random_number.hpp"
 
 namespace silkworm {
 
-BlockNum BodySequence::highest_block_in_db() { return highest_block_in_db_; }
+seconds_t BodySequence::kRequestDeadline;
+BlockNum BodySequence::kMaxBlocksPerMessage;
+size_t BodySequence::kPerPeerMaxOutstandingRequests;
+milliseconds_t BodySequence::kNoPeerDelay;
+
+BodySequence::BodySequence(const Db::ReadOnlyAccess& dba, const ChainIdentity& ci)
+    : db_access_(dba), chain_identity_(ci) {
+    recover_initial_state();
+}
+
+BodySequence::~BodySequence() {
+
+}
+
+void BodySequence::recover_initial_state() {
+    // does nothing
+}
```

Comment:
I would to maintains it to remember the point where the BodyPersistence object can be safely destroyed.
If in the final version it will be still an empty method I will delete it.